core/src/main/java/fr/sncf/osrd/utils/graph/Pathfinding.java

```diff
@@ -74,6 +80,10 @@ public record Range(long start, long end){}
      * Used in STDCM. Either totalDistanceUntilEdgeLocation or edgeRangeCost must be defined.
     */
     private TotalCostUntilEdgeLocation<EdgeT> totalCostUntilEdgeLocation = null;
+    /**
+     * Timeout, in seconds, to avoid infinite loop when no path can be found.
+     */
+    private Long timeout = null;
```

Comment:
We could set the default value to TIMEOUT, and then let the caller override it when relevant (which should only happen occasionally in performance tests anyway). There's more pathfinding calls than the ones that have been edited in this PR.